roles/etcd/defaults/main.yaml

```diff
@@ -12,11 +12,15 @@ l_etcd_static_pod: "{{ not (r_etcd_common_skip_command_shim is defined and r_etc
 r_etcd_common_etcd_runtime: "{{ 'runc' if l_is_etcd_system_container else ('static_pod' if l_etcd_static_pod else ('docker' if openshift_is_containerized else 'host')) }}"
 
 r_etcd_default_version: "3.2.15"
-osm_etcd_image: "registry.access.redhat.com/rhel7/etcd:{{ r_etcd_upgrade_version | default(r_etcd_default_version) }}"
+# lib_utils_oo_etcd_image is a custom filter defined in roles/lib_utils/filter_plugins/oo_filters.py
+# This filter attempts to combine oreg_url host with project/component from etcd_image_dict.
+# "oreg.example.com/openshift3/ose-${component}:${version}"
+# becomes "oreg.example.com/rhel7/etcd:{{ r_etcd_upgrade_version | default(r_etcd_default_version) }}"
+osm_etcd_image: "{{ etcd_image_dict[openshift_deployment_type] | lib_utils_oo_etcd_image((oreg_url | default('None')) }}"
 etcd_image_dict:
   origin: "quay.io/coreos/etcd:v{{ r_etcd_upgrade_version | default(r_etcd_default_version) }}"
-  openshift-enterprise: "{{ osm_etcd_image }}"
-etcd_image: "{{ etcd_image_dict[openshift_deployment_type] }}"
+  openshift-enterprise: "registry.access.redhat.com/rhel7/etcd:{{ r_etcd_upgrade_version | default(r_etcd_default_version) }}"
+etcd_image: "{{ osm_etcd_image }}"
```

Comment:
We specify setting osm_etcd_image in the hosts.example; for origin, setting this value would previously have no effect.  This cleans up the logic.